src/framework/ui/view/interactiveprovider.cpp

```diff
@@ -175,19 +178,36 @@ Ret InteractiveProvider::showProgress(const std::string& title, Progress* progre
 RetVal<io::path_t> InteractiveProvider::selectOpeningFile(const std::string& title, const io::path_t& dir,
                                                           const std::vector<std::string>& filter)
 {
-    return openFileDialog(FileDialogType::SelectOpenningFile, title, dir, filter);
+    RetVal<InteractiveProvider::FileInfo> fileInfo = openFileDialog(FileDialogType::SelectOpenningFile, title, dir, filter);
+    return fileInfo.ret ? RetVal<io::path_t>::make_ok(fileInfo.val.path) : fileInfo.ret;
 }
 
 RetVal<io::path_t> InteractiveProvider::selectSavingFile(const std::string& title, const io::path_t& path,
                                                          const std::vector<std::string>& filter,
                                                          bool confirmOverwrite)
 {
-    return openFileDialog(FileDialogType::SelectSavingFile, title, path, filter, confirmOverwrite);
+    RetVal<InteractiveProvider::FileInfo> fileInfo
+        = openFileDialog(FileDialogType::SelectSavingFile, title, path, filter, confirmOverwrite);
+
+    if (!fileInfo.ret) {
+        return fileInfo.ret;
+    }
+
+    std::string suffix = muse::io::suffix(fileInfo.val.path);
+    if (muse::contains(fileInfo.val.allowedExtensions, suffix)
+        || muse::contains(fileInfo.val.allowedExtensions, ALL_FILES_FILTER)
+        || muse::contains(fileInfo.val.allowedExtensions, WINDOWS_NO_EXTENSION_FILTER)) {
```

Comment:
Perhaps we made a mistake here: instead of a filter, we will see an extension. So maybe the logic should be `muse::contains(fileInfo.val.allowedExtensions, std::string())`, to check if an empty extension is one of the allowed extensions. Because in that case, we should not add an extra `.`.